docs/yaml/functions/response_file.yaml

```diff
@@ -0,0 +1,75 @@
+name: response_file
+returns: list[str] | str
+description: |
+  Write argument list to a response file, if the resulting
+  length of the command line is longer than a maximum value.
+  A response file is a mechanism allowing a program to receive
+  more arguments than what would be allowed by the command line. See
+  [msvc compiler documentation](https://learn.microsoft.com/en-us/cpp/build/reference/at-specify-a-compiler-response-file?view=msvc-170)
+  as an example.
+
+  Returns either the list of arguments or the path to the response
+  file.
+
+  This function is designed to be used with `custom_target`,
+  `run_target`, `run_command`, `test`, or any other function
+  running a custom executable or script.
+
+varargs:
+  name: arguments
+  type: str | file | list
+  description: The argument list.
+
+kwargs:
+  prefix:
+    type: str
+    default: "'@'"
+    description: |
+      The prefix to prepend to the response file path.
+      Use an empty string if you don't need a prefix.
+  
+  quote:
+    type: str | bool
+    default: true
+    description: |
+      If `true`, quote the arguments in the response file
+      using the default quote function, based on the operating
+      system.
+
+      If `false`, do not quote the arguments.
+
+      Can also be `'sh'` to use sh-like quoting,
+      `'cmd'` for Windows cmd-like quoting,
+      or `'gcc'` for gcc-like quoting.
+
+  max_args_length:
+    type: int | bool
+    default: 127
+    description: |
+      Maximum allowed command line length, after joining and
+      quoting the arguments. It doesn't take into account the
+      length of the command itself, or the length of other
+      arguments that could be provided to the comamnd.
+
+      If `true` or `0`, a response file is always created,
+      no matter the length of the command line.
+
+      If `false`, the function do not create a response file
+      but returns the list of arguments.
+  
+  separator:
+    type: str
+    default: "' '"
+    description: |
+      The separator used to join the arguments. Usually a
+      space character, but could also be a `'\n'` to put
+      one argument per line in the response file.
```

Comment:
Is this really needed?

Reply:
Yes. For instance, python `argparse` expects one argument per line.

Reply:
TIL argparse can do that, indeed: https://docs.python.org/3/library/argparse.html#fromfile-prefix-chars